CONTRIBUTING.rst

```diff
@@ -1,37 +1,214 @@
 
-Thanks for your interest in contributing to ``aiobotocore``, there are multiple
-ways and places you can contribute.
+  8. Make pull request from github page for your branch against master branch
 
-Fist of all just clone repository::
+  9. Optionally make backport pull Request(s) for landing a bug fix
+     into released aiobotocore versions.
 
-    $ git clone git@github.com:aio-libs/aiobotocore.git
+  10. Add changes in file ``CHANGES.txt`` for this read (`Changelog update`_).
 
-Create virtualenv with at least python3.5 (older version are not supported).
-For example using *virtualenvwrapper* commands could look like::
+
+Preconditions for running aiobotocore test suite
+------------------------------------------------
+
+We expect you to use a python virtual environment to run our tests.
+
+There are several ways to make a virtual environment.
+
+If you like to use *virtualenv* please run:
+
+.. code-block:: shell
 
    $ cd aiobotocore
-   $ mkvirtualenv --python=`which python3.5` aiobotocore
+   $ virtualenv -p /usr/bin/python3 venv-aiobotocore
+   $ . venv-aiobotocore/bin/activate
+
+For standard python *venv-aiobotocore*:
+
+.. code-block:: shell
+
+   $ cd aiobotocore
+   $ python3 -m venv venv-aiobotocore
+   $ . venv-aiobotocore/bin/activate
+
+For *virtualenvwrapper*:
+
+.. code-block:: shell
+
+   $ cd aiobotocore
+   $ mkvirtualenv -p /usr/bin/python3 venv-aiobotocore
+   $ workon venv-aiobotocore
+
+There are other tools like *pyvenv* but you know the rule of thumb
+now: create a python3 virtual environment and activate it.
+
+After that please install libraries required for development:
+
+.. code-block:: shell
+
+   $ pip install -r requirements/dev.txt
+
+Congratulations, you are ready to run the test suite!
+
+Run aiobotocore test suite
+--------------------------
+
+After all the preconditions are met you can run tests typing the next
+command:
+
+.. code-block:: shell
+
+   $ make test
+
+Run aiobotocore flake suite
+---------------------------
+
+We try our best to write code according to pep, it will allow you to check yourself.
+
+.. code-block:: shell
+
+   $ make flake
+
+Run aiobotocore coverage suite
+------------------------------
+
+We are trying hard to have good test coverage; please don't make it worse.
+
+.. code-block:: shell
+
+   $ make cov
+
+.. note::
+
+  If you want to use AWS within the test suite, you need create config ``~/.aws/credentials`` or the ``environment variable``
+
+  Then the test run will look like this:
+  .. code-block:: shell
+
+     $ make aws-cov
+     $ make aws-test
+
+
+Documentation
+-------------
+
+We encourage documentation improvements.
+
+Please before making a Pull Request about documentation changes run:
+
+.. code-block:: shell
+
+   $ make doc
+
+Once it finishes it will output the index html page
+``open file://`pwd`/docs/_build/html/index.html``.
+
+Go to the link and make sure your doc changes looks good.
+
+Commit message format
```

Comment:
@jettify do we want to get this pedantic of naming your branches and formatting your commit messages in particular ways??  I think this needs to be discussed and not included in a PR like this that does a lot of things.

Reply:
Good point. I propose then to make a separate discussion. just well understand what is happening with the project, and the commit name should help in this